### PR TITLE
DEX-637 Backward compatibility tests for database

### PR DESCRIPTION
**Pull request by vsuharnikov**
Integration tests:
* Added DatabaseBackwardCompatTestSuite;
* The common part for backward compat tests of MultipleDifferentMatchersTestSuite is separated to BackwardCompatSuiteBase;
* MultipleDifferentMatchersTestSuite is renamed to OrderBookBackwardCompatTestSuite;

Test API:
* Annotations: removed Target=ElementType.METHOD;
* BaseContainer: fixed a bug with printing debug messages when a container wasn't started;

### CODE REVIEW

File: dex-it/src/test/scala/com/wavesplatform/it/sync/compat/DatabaseBackwardCompatTestSuite.scala [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    orders.foreach(dex2.api.place)
+
+    dex2.api.waitForOrder(orders.last)(_.status != OrderStatus.NotFound)
+    dex2.stopWithoutRemove()
+
+    dex1.start()
+    orders.groupBy(_.assetPair).valuesIterator.foreach { orders =>
+      dex1.api.waitForOrder(orders.last)(_.status != OrderStatus.NotFound)
+    }
+    Thread.sleep(3.seconds.toMillis) // An additional time to wait the concurrent processing
```
Comment (DenisSerebryansky) [inline dex-it/src/test/scala/com/wavesplatform/it/sync/compat/DatabaseBackwardCompatTestSuite.scala:32]: Lets make implicit conversion of finite duration to long in `MatcherSuiteBase`

File: dex-it/src/test/scala/com/wavesplatform/it/sync/compat/DatabaseBackwardCompatTestSuite.scala [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+  override protected def beforeAll(): Unit = {
+    super.beforeAll()
+
+    val containerDataDir = DexContainer.containerPath("data")
+    List(dex1, dex2).foreach {
+      _.underlying.configure {
+        _.withFileSystemBind(localLogsDir.resolve("db").toString, containerDataDir, BindMode.READ_WRITE)
+      }
+    }
```
Comment (DenisSerebryansky) [inline dex-it/src/test/scala/com/wavesplatform/it/sync/compat/DatabaseBackwardCompatTestSuite.scala:47]: If I got it correct, you bind local directory with the directory inside DEX containers. Why do we need this?

Comment (vsuharnikov) [reply dex-it/src/test/scala/com/wavesplatform/it/sync/compat/DatabaseBackwardCompatTestSuite.scala:47]: To share the database between them:
1. We place orders on the old DEX version
2. We stop the old DEX version
3. We start the new DEX version with the same database and make checks

File: dex-it/src/test/scala/com/wavesplatform/it/sync/compat/DatabaseBackwardCompatTestSuite.scala (hunk 1 above)
Comment (vsuharnikov) [reply dex-it/src/test/scala/com/wavesplatform/it/sync/compat/DatabaseBackwardCompatTestSuite.scala:32]: Created a task for this

**Pull request merged**